src/sensor.h

```diff
@@ -9,6 +9,7 @@
 #include "source.h"
 #include "core/extension.h"
 #include "proc/formats-converter.h"
+#include "core/features/feature-interface.h"
```

Comment:
Why do you need this if `supports_feature` doesn't return it? Why doesn't it return it?

Reply:
I can forward declare it and remove the include if you want. Since it is a small interface I thought it won't include many other files and so will not affect compilation time.

Reply:
I meant that it's not in use here.
1. Why not?
2. If we don't want to use it here, then don't need the include...

Reply:
Ahh... I understand now. A left over from an implementation where `supports_feature` received a parameter of type `feature_interface *`. I will remove.

Reply:
But what if we want anything that supports "auto-exposure" to have an API called "get_actual_exposure()"?

Reply:
Removed include

Reply:
> But what if we want anything that supports "auto-exposure" to have an API called "get_actual_exposure()"?

We use options for that

Reply:
Very sad that we do...
But you're missing my point.
A feature should enable additional functionality. That's why we have a feature interface: to expose additional functionality that it not available in the base feature implementer. If all it ever does is enable an option, then we haven't done anything because we can check if the option exists...
The interface only makes sense if we actually return it. How else would it be used?

Reply:
Maybe the name `feature` is too extensive here. The purpose of this PR is to allow code that is common to multiple camera models know if some "things" should be enabled based on run time information like camera model and device FW version.
The PR purpose is not to expose functionality, just defer the enablement logic from a common place to the relevant device.

Reply:
I view features the same as extensions, able to do what you said but also do what extensions do, just internally (at least for now) and with more detail. They don't *have* to expose functionality, but should be able to.
Also, after thinking about it, I'm not sure with limiting this to the sensor -- features should be at the product level, not sensor.
Let's talk offlne.

Reply:
After our discussion, feature availability is decided at device level.
Some features have implementation for their logic